beginner_source/dist_overview.rst

```diff
@@ -15,205 +15,57 @@ to the technology that can best serve your use case.
-`torch.distributed.rpc <https://pytorch.org/docs/stable/rpc.html>`__ aims at
-supporting general distributed training scenarios.
-
-`torch.distributed.rpc <https://pytorch.org/docs/stable/rpc.html>`__
-has four main pillars:
-
-* `RPC <https://pytorch.org/docs/stable/rpc.html#rpc>`__ supports running
-  a given function on a remote worker.
-* `RRef <https://pytorch.org/docs/stable/rpc.html#rref>`__ helps to manage the
-  lifetime of a remote object. The reference counting protocol is presented in the
-  `RRef notes <https://pytorch.org/docs/stable/rpc/rref.html#remote-reference-protocol>`__.
-* `Distributed Autograd <https://pytorch.org/docs/stable/rpc.html#distributed-autograd-framework>`__
-  extends the autograd engine beyond machine boundaries. Please refer to
-  `Distributed Autograd Design <https://pytorch.org/docs/stable/rpc/distributed_autograd.html#distributed-autograd-design>`__
-  for more details.
-* `Distributed Optimizer <https://pytorch.org/docs/stable/rpc.html#module-torch.distributed.optim>`__
-  automatically reaches out to all participating workers to update
-  parameters using gradients computed by the distributed autograd engine.
-
-RPC Tutorials are listed below:
-
-1. The `Getting Started with Distributed RPC Framework <../intermediate/rpc_tutorial.html>`__
-   tutorial first uses a simple Reinforcement Learning (RL) example to
-   demonstrate RPC and RRef. Then, it applies a basic distributed model
-   parallelism to an RNN example to show how to use distributed autograd and
-   distributed optimizer.
-2. The `Implementing a Parameter Server Using Distributed RPC Framework <../intermediate/rpc_param_server_tutorial.html>`__
-   tutorial borrows the spirit of
-   `HogWild! training <https://people.eecs.berkeley.edu/~brecht/papers/hogwildTR.pdf>`__
-   and applies it to an asynchronous parameter server (PS) training application.
-3. The `Distributed Pipeline Parallelism Using RPC <../intermediate/dist_pipeline_parallel_tutorial.html>`__
-   tutorial extends the single-machine pipeline parallel example (presented in
-   `Single-Machine Model Parallel Best Practices <../intermediate/model_parallel_tutorial.html>`__)
-   to a distributed environment and shows how to implement it using RPC.
-4. The `Implementing Batch RPC Processing Using Asynchronous Executions <../intermediate/rpc_async_execution.html>`__
-   tutorial demonstrates how to implement RPC batch processing using the
-   `@rpc.functions.async_execution <https://pytorch.org/docs/stable/rpc.html#torch.distributed.rpc.functions.async_execution>`__
-   decorator, which can help speed up inference and training. It uses
-   RL and PS examples similar to those in the above tutorials 1 and 2.
-5. The `Combining Distributed DataParallel with Distributed RPC Framework <../advanced/rpc_ddp_tutorial.html>`__
-   tutorial demonstrates how to combine DDP with RPC to train a model using
-   distributed data parallelism combined with distributed model parallelism.
-
-
```

Comment:
Also, suggesting to add more links in the end to the existing tutorials and other sources that would help users to get started. Could add under the See Also section.

Reply:
yea, that's a good point.  i'm actually not all that familiar with the tutorials content we already have, but i suspect a lot of it is out of date.  I may do a search for any important ones to add now, but i think we can also build that up more over time as we add more tutorials or update old ones.